admin/artists.php

```diff
@@ -4,181 +4,247 @@ function gigpress_artists() {
-<?php endif; ?>
+				</thead>
+				<tbody class="gigpress-artist-sort">
+				<?php
+
+				if ( $artists ) {
+
+					$i = 0;
+					foreach ( $artists as $artist ) {
+
+						if ( $n = $wpdb->get_var( "SELECT count(*) FROM " . GIGPRESS_SHOWS . " WHERE show_artist_id = " . $artist->artist_id . " AND show_status != 'deleted'" ) ) {
+							$count = '<a href="' . admin_url( 'admin.php?page=gigpress-shows&amp;artist_id=' . $artist->artist_id ) . '">' . $n . '</a>';
+						} else {
+							$count = 0;
+						}
+
+						$i ++;
+						$style = ( $i % 2 ) ? '' : ' class="alternate"';
+						// Print out our rows.
+						?>
+						<tr<?php echo $style; ?> id="artist_<?php echo $artist->artist_id; ?>">
+							<td class="gp-tiny">
+								<?php if (
+									isset( $pagination )
+									|| isset( $_GET['orderby'] )
+									   && (
+										   'artist_id' == $_GET['orderby']
+										   || 'artist_name' == $_GET['orderby']
+									   )
+								) :
+									echo $artist->artist_order;
+								else: ?>
+									<img src="<?php echo esc_url( GIGPRESS_PLUGIN_URL . 'images/sort.png' ); ?>" alt="" class="gp-sort-handle" />
+								<?php endif; ?>
+							</td>
+							<td class="gp-tiny"><?php echo $artist->artist_id; ?></td>
+							<td><?php if ( ! empty( $artist->artist_url ) ) {
+									echo '<a href="' . esc_url( $artist->artist_url ) . '">';
+								}
+								echo wptexturize( $artist->artist_name );
+								if ( ! empty( $artist->artist_url ) ) {
+									echo '</a>';
+								} ?></td>
+							<td class="gp-centre"><?php echo $count; ?></td>
+							<td class="gp-centre">
+								<a href="<?php echo admin_url( 'admin.php?page=gigpress-artists&amp;gpaction=edit&amp;artist_id=' . $artist->artist_id . $url_args ); ?>" class="edit">
+									<?php _e( "Edit", "gigpress" ); ?>
+								</a>
+								<?php if ( ! $count ) { ?> |
+									<a href="<?php echo wp_nonce_url( admin_url( 'admin.php?page=gigpress-artists&amp;gpaction=delete&amp;artist_id=' . $artist->artist_id . $url_args ), 'gigpress-action' ); ?>" class="delete">
+										<?php _e( "Delete", "gigpress" ); ?>
+									</a>
+								<?php } ?>
+							</td>
+						</tr>
+					<?php }
+				} else {
+
+					// We don't have any artists, so let's say so
+					?>
+					<tr>
+						<td colspan="5"><strong><?php _e( "No artists in the database", "gigpress" ); ?></strong></td>
+					</tr>
+				<?php } ?>
+				</tbody>
+				<tfoot>
+				<tr>
+					<th scope="col" class="gp-tiny">&nbsp;</th>
+					<th scope="col" class="gp-tiny">ID</th>
+					<th scope="col"><?php _e( "Artist name", "gigpress" ); ?></th>
+					<th scope="col" class="gp-centre"><?php _e( "Number of shows", "gigpress" ); ?></th>
+					<th class="gp-centre" scope="col"><?php _e( "Actions", "gigpress" ); ?></th>
+				</tr>
+				</tfoot>
+			</table>
+
+			<?php if ( isset( $pagination ) ) : ?>
+				<div class="tablenav">
+					<?php
+					if ( $gpo['artist_pagination'] ) {
+						echo $pagination['output'];
+					}
```

Comment:
Typically, within templates, we prefer to use `endif;` style conditionals / loops.

```suggestion
					if ( $gpo['artist_pagination'] ) :
						echo $pagination['output'];
					endif;
```